python/tank/platform/application.py

```diff
@@ -43,6 +43,12 @@ def __init__(self, engine, descriptor, settings, instance_name, env):
         self.__engine = engine
         self.__instance_name = instance_name
 
+        # By default we will assume that an app does not support context
+        # changes. When changing contexts, the engine and all active apps
+        # will need to explicitly support the switch. If any do not, the
+        # engine will be torn down and restarted.
+        self._context_change_allowed = False
```

Comment:
that would further emphasize that this is not a state that would change at runtime but more of a fundamental constant; either an app supports this or it doesn't and it won't change over time.